### DISCUSSION TITLE
v4.13.2 - Track OrgLimits data on Log__c

### DISCUSSION THREAD

**Discussion by jongpie** [jongpie/NebulaLogger#647] [Announcements]
#  🎉 🪵 💯  Nebula Logger's 100th Release! 💯 🪵  🎉  

Yet another neat milestone for the project 🥳

# Core Unlocked Package Changes

- Resolved #630 by adding new functionality to automatically track organization limits data, returned by [the class `System.OrgLimits`](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_System_OrgLimits.htm)
  - This data is automatically stored as JSON in the new field `Log__c.OrganizationLimits__c`
  - You can disable storing this data, using the new `LoggerParameter__mdt` record `StoreOrganizationLimits`
  - This data is displayed within a new "Organization Limits" tab on the `Log__c` flexipage, using a new LWC `logOrganizationLimits` (shown below). The design of this LWC is intended to mimic the existing formula fields shown on `LogEntry__c` that are used to track transaction limits.

    ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/e7473fc8-85c4-48b3-8ddb-fc0f044ec355)
    
      - For comparison, this image shows the existing formula fields on `LogEntry__c` that are used to track transaction limits.
        
        ![image](https://github.com/jongpie/NebulaLogger/assets/1267157/d109ddf2-9734-4a05-a743-cdb543228c66)

- Added another new `LoggerParameter__mdt` record, `StoreTransactionLimits`, to control if transaction limits are retrieved from [the class `System.Limits`](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_limits.htm) & stored on `LogEntry__c`
  - Previously, transaction limits were always populated, with no way to disable it - using the `System.Limits` class does seem to have some overhead (namely, it uses CPU time), so being able to disable this functionality can, hilariously, help to reduce transaction limits usage.
- `loggerPageSection` LWC changes: Fixed some padding in `loggerPageSection.html` so it's more consistent with 'real' field section headers

# Installation Info

## Core Unlocked Package - no namespace

Full Changelog: https://github.com/jongpie/NebulaLogger/compare/v4.13.1...v4.13.2

- SF CLI: `sf package install --wait 20 --security-type AdminsOnly --package 04t5Y000001MkEmQAK`
- SFDX CLI: `sfdx force:package:install --wait 20 --securitytype AdminsOnly --package 04t5Y000001MkEmQAK`
- Sandbox: https://test.salesforce.com/packaging/installPackage.apexp?p0=04t5Y000001MkEmQAK
- Production: https://login.salesforce.com/packaging/installPackage.apexp?p0=04t5Y000001MkEmQAK

<hr /><em>This discussion was created from the release <a href='https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.2'>Track OrgLimits data on Log__c</a>.</em>

**Comment by jamessimone** [comment]
Aw yeah! Congratulations!

**Reply by jongpie** [reply]
Woohoo! Thanks again for your help with this one, and all the other releases you've helped with!
